bluetooth/h4_protocol.cc

```diff
@@ -132,9 +144,98 @@ size_t H4Protocol::Send(uint8_t type, const uint8_t* data, size_t length){
     return ret;
 }
 
+bool H4Protocol::IsIntelController(uint16_t vid, uint16_t pid) {
+
+	if((vid == INTEL_VID) && ((pid == INTEL_PID_8265) ||
+				(pid == INTEL_PID_3168)))
+            return true;
+	else
+	    return false;
+}
+
+void H4Protocol::GetUsbpath(void) {
+
+    size_t count, i;
+    int ret, busnum, devnum;
+    struct libusb_device **dev_list = NULL;
+    struct libusb_context *ctx;
+    uint16_t vid = 0, pid = 0;
+
+    ALOGD(" Initializing GenericUSB (libusb-1.0)...\n");
+    ret = libusb_init(&ctx);
+    if (ret < 0) {
+        ALOGE("libusb failed to initialize: %d\n", ret);
+        return;
+    }
+    count = libusb_get_device_list(ctx, &dev_list);
+    if (count <= 0) {
+        ALOGE("Error getting USB device list: %s\n", strerror(count));
+        goto exit;
+    }
+    for (i = 0; i < count; ++i) {
+        struct libusb_device* dev = dev_list[i];
+        busnum = libusb_get_bus_number(dev);
+        devnum = libusb_get_device_address(dev);
+        struct libusb_device_descriptor descriptor;
+        ret = libusb_get_device_descriptor(dev, &descriptor);
+        if (ret < 0)  {
+            ALOGE("Error getting device descriptor %d ",ret);
+	    libusb_free_device_list(dev_list,count);
+            goto exit;
+        }
+        vid = descriptor.idVendor;
+        pid = descriptor.idProduct;
+        if (H4Protocol::IsIntelController(vid,pid)) {
+            sprintf(dev_address,"/dev/bus/usb/%03d/%03d",busnum,devnum);
```

Comment:
- introduce space between , and "
- introduce space between , and busnum
- introduce space between , and devnum